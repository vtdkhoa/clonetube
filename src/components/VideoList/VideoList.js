import React, { Fragment } from 'react'
import SideBar from '../../containers/SideBar/SideBar'
import InfiniteScroll from '../InfiniteScroll/InfiniteScroll'
import VideoPreview from '../VideoPreview/VideoPreview'
import './VideoList.scss'

const VideoList = props => {
  const videoPreviews = getVideoPreviews(props.videos)

  return (
    <Fragment>
      <SideBar/>
      <div className="video-list">
        <InfiniteScroll
          bottomReachedCallback={props.bottomReachedCallback}
          showLoader={props.showLoader}
        >
          {videoPreviews}
        </InfiniteScroll>
      </div>
    </Fragment>
  )
}

const getVideoPreviews = videos => {
  if (!videos || !videos.length) {
    return null
  }
  return videos.map(video => (
    <VideoPreview
      horizontal={true}
      expanded={true}
      video={video}
      key={video.id}
      pathname={'/watch'}
      search={`?=${video.id}`}
    />
  ))
}

export default VideoList