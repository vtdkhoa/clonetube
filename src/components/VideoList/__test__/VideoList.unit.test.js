import React from 'react'
import VideoList from '../VideoList'
import { shallow } from 'enzyme'

const mockUpVideos = [
  {
    id: 'videoId',
    snippet: {
      description: 'video description'
    }
  }
]

describe('VideoList', () => {
  test('renders without props', () => {
    const wrapper = shallow(<VideoList/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with videos props', () => {
    const wrapper = shallow(
      <VideoList videos={mockUpVideos}/>
    )
    expect(wrapper).toMatchSnapshot()
  })
})