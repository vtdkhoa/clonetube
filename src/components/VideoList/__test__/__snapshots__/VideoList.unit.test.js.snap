// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList renders with videos props 1`] = `
<Fragment>
  <SideBar />
  <div
    className="video-list"
  >
    <InfiniteScroll>
      <VideoPreview
        expanded={true}
        horizontal={true}
        key="videoId"
        pathname="/watch"
        search="?v=videoId"
        video={
          Object {
            "id": "videoId",
            "snippet": Object {
              "description": "video description",
            },
          }
        }
      />
    </InfiniteScroll>
  </div>
</Fragment>
`;

exports[`VideoList renders without props 1`] = `
<Fragment>
  <SideBar />
  <div
    className="video-list"
  >
    <InfiniteScroll />
  </div>
</Fragment>
`;
